x-pack/plugin/inference/src/main/plugin-metadata/entitlement-policy.yaml

```diff
@@ -1,2 +1,10 @@
 com.google.api.client:
   - set_https_connection_properties
+software.amazon.awssdk.profiles:
+  - files: # AWS Clients always try to access the credentials and config files, even if we configure otherwise
+    - relative_path: .aws/credentials
```

Comment:
Could you please double check if this uses a different path on Windows?

Reply:
oh good catch!

Reply:
According to [these docs](https://docs.aws.amazon.com/powershell/latest/userguide/specifying-your-aws-credentials.html#specifying-your-aws-credentials-store), on Windows, it's `C:\Users\username\.aws\credentials`, and so I think the path `.aws/credentials` relative to `home` should still work.

Reply:
...oh, and come to think of it... this is the same setting we used in the SM security policy, so I suppose it can't be worse than status quo.

Reply:
I think this should be done in a consistent way with `repository-s3`, adapting here or there

Reply:
True. That one adds all of `.aws`. That seems reasonable to me, though it would differ from what inference's current SM policy contains.

It seems inherently reasonable to me that we'd grant all of `.aws` to the `software.amazon.awssdk.profiles`; it also seems reasonable that we'd leave this alone so it matches the SM policy. (Ironically, the `repository-s3` plugin is not modular, so it grants `.aws` to `ALL_UNNAMED`, which seems less desirable.)

I think I'll leave this one as-is so it matches its own SM policy file. We can change the `repository-s3` one in a separate PR if it's the case that `.aws/credentials` and `.aws/config` are sufficient.

Reply:
This is captured in ES-11022 now.

Reply:
> I think this should be done in a consistent way with repository-s3, adapting here or there

Note that there's a difference in that `repository-s3` uses the AWS v1 SDK which can cope with a `SecurityException` here, whereas inference uses the AWS v2 SDK that cannot. This is a bug in the v2 SDK, see https://github.com/aws/aws-sdk-java-v2/issues/5754. When AWS release the fixed version we can upgrade it and withdraw these permissions here too.

Reply:
++ we will keep parity with SM policies for v9 GA, so these will stay here.